crates/wdk-build/src/lib.rs

```diff
@@ -605,116 +714,269 @@ impl Config {
+            // standalone, this scenario is treated as a warning.
+            tracing::warn!("No WDK configurations detected.");
+            // check_cfg must be emitted even if no WDK configurations are detected, so that
+            // cfg options are still checked
+            Config::emit_check_cfg_settings();
+            Ok(())
+        }
+
+        Err(error) => Err(error),
+    }
+}
+
+/// Configure a Cargo build of a library that depends on the WDK, then execute a
+/// function or closure with the [`Config`] derived from `metadata.wdk` sections
+/// of `Cargo.toml`s.
+///
+/// This emits specially formatted prints to Cargo based on the [`Config`]
+/// derived from `metadata.wdk` sections of `Cargo.toml`s.
+///
+/// Cargo build graphs that have no valid WDK configurations will emit a
+/// warning, but will still return [`Ok`]. This allows libraries
+/// designed for multiple configurations to sucessfully compile when built in
+/// isolation.
+///
+/// # Errors
+///
+/// This function will return an error if the [`Config`] fails to be
+/// serialized
+pub fn configure_wdk_library_build_and_then<F, E>(mut f: F) -> Result<(), E>
+where
+    F: FnMut(Config) -> Result<(), E>,
+    E: std::convert::From<ConfigError>,
+{
+    match Config::from_env_auto() {
+        Ok(config) => {
+            config.configure_library_build()?;
+            Ok(f(config)?)
+        }
+        Err(ConfigError::TryFromCargoMetadataError(
+            TryFromCargoMetadataError::NoWdkConfigurationsDetected,
+        )) => {
+            // No WDK configurations will be detected if the crate is not being used in a
+            // driver. Since this is usually the case when libraries are being built
+            // standalone, this scenario is treated as a warning.
+            tracing::warn!("No WDK configurations detected.");
+            // check_cfg must be emitted even if no WDK configurations are detected, so that
+            // cfg options are still checked
+            Config::emit_check_cfg_settings();
+            Ok(())
+        }
+
+        Err(error) => Err(error.into()),
+    }
+}
+
+/// Configure a Cargo build of a binary that depends on the WDK using a
+/// [`Config`] derived from `metadata.wdk` sections of `Cargo.toml`s.
+///
+/// # Errors
+///
+/// This function will return an error if:
+/// * any of the required WDK paths do not exist
+/// * the C runtime is not configured to be statically linked
+///
+/// # Panics
+///
+/// Panics if the invoked from outside a Cargo build environment
+pub fn configure_wdk_binary_build() -> Result<(), ConfigError> {
+    Config::from_env_auto()?.configure_binary_build()
+}
+
+// This currently only exports the driver type, but may export more metadata in
+// the future. `EXPORTED_CFG_SETTINGS` is a mapping of cfg key to allowed cfg
+// values
+lazy_static::lazy_static! {
+    // FIXME: replace lazy_static with std::Lazy once available: https://github.com/rust-lang/rust/issues/109736
```

Comment:
Lazy is now stable if I'm not mistaken.  Not something that needs to be fixed right this instant though.

Reply:
> Lazy is now stable if I'm not mistaken. Not something that needs to be fixed right this instant though.

Its available in stable as of 1.80.0 (released 5 days ago). Although not formally stated (or enforced) I've been trying to keep support for Stable and Stable-1. 

There are also other places in the repo where lazy_static is used, that aren't supported by std::sync::Lazy. (ex. LazyLock is not available in no_std, so in those situations, we should migrate to once_cell w/ the `racy` feature)

Reply:
tracking this in https://github.com/microsoft/windows-drivers-rs/issues/192